### ISSUE TITLE
ArgNull error on login

### ISSUE THREAD

**Issue by lutra333** [mtaheij/PoGo-UWP#55]
Device info
------------------
- Phone type: Lumia 950
- App version 1.3.1

Steps to reproduce
----------------
Open app. enter valid hash code, press 'store and continue'

Actual behavior
-----------------
Getting error:

Arg_NullReferenceException

Expected behavior
---------------
Successful login

Other information
---------------
Same behavior from the version 1.2.9 and up

key is valid (tried diferent pokehash numbers including brand new one)

What else I tried:

Delete app, restart, download app again, install app - not working. getting same error
Multiply exits and loading the app - not working. getting same error
Delete app, factory reset phone, restore phone, download app, install app - same behavior (error)
New pokehash - not working, getting error

**Comment by DarkLegend481** [comment]
Did you tried 1.3.2? I got the same error on 1.3.1...

**Comment by DarkLegend481** [comment]
Oh, 1.3.1 uses 0.57.2 API, and 1.3.2 uses the 0.61.0 API... Maybe, it's a API error...

**Comment by lutra333** [comment]
Yes, i have same problem with 1.3.2 (i edited the issue accordingly)

**Comment by DarkLegend481** [comment]
Did you tried uninstall PoGo, restart your Lumia and install PoGo again?

**Comment by DarkLegend481** [comment]
With 1.3.2, obviously...

**Comment by lutra333** [comment]
Yes. i tried all the steps described above (including factory reset) - and still getting same error

**Comment by DarkLegend481** [comment]
Now, I get this error:
![wp_ss_20170409_0002](https://cloud.githubusercontent.com/assets/22161127/24837656/6e823510-1d0f-11e7-9bc9-0d1f2bc5ee76.png)

**Comment by lutra333** [comment]
I still have Arg_NullReferenceException error

**Comment by danmartinez16** [comment]
I'm getting the same error: Failed parsing pokehash response headers

**Comment by DarkLegend481** [comment]
Any fix @mtaheij?

**Comment by mtaheij** [comment]
I do not have this problem, so reproducing it will be hard for me, let alone fixing it :-S

**Comment by Xiggles** [comment]
Installed the app yesterday and until 3 hours ago the app still worked perfectly. 
But now I'm getting the same error as above. Arg_NullReferenceException. Version 1.3.2

**Comment by lutra333** [comment]
Just in case^ switching from Wifi to 4G doesn't help me. Still getting Arg_Null error, doesnt matter if on wifi or not,

**Comment by WanderingJesusFreak** [comment]
This is one of the most frustrating errors, because it is impossible to reproduce. For instance, I went for a walk last night. An hour before the walk, everything worked fine. At the beginning of the walk, I got the error. So, my wife played on her phone and we walked and talked while I went through all the uninstall, reboot, reinstall, turn on and off various functions... and nothing. Then, 30 minutes into our walk, it started working again, for no reason.

**Comment by andremvm** [comment]
Could this error be due to Bossland Tripwire service ( https://wiki.thebuddyforum.com/index.php?title=BosslandGmbH:HelpDesk:What_is_Tripwire%3F )? Maybe that is why it is so "random", whenver they activate this security feature we cannot login until it is de-activated again; this could also explain why our accounts are flagged by Niantic but not banned yet...

**Comment by Rokuzer** [comment]
I hace the dame problem failed parsing pokehash response headers

**Comment by DarkLegend481** [comment]
To solve the "failed parsing pokehash response headers" issue, you need update your key to a 0.61 API key (or something like that)

**Comment by DarkLegend481** [comment]
To anyone that doesn't have a key: dir50/users/incacode

Youtube Channel (Corpright): youtube.com/channel/UC0Ugx4PLK44JRelI6XUvvIA

**Comment by DarkLegend481** [comment]
I'm tried one of their keys and it's working!

**Comment by Xiggles** [comment]
Not sure if that's the case. I've gotten the error message a few times now, but it always seems to go away after around 15 minutes.

**Comment by DarkLegend481** [comment]
For me, after putting the new key, PoGo started working fine...

**Comment by WanderingJesusFreak** [comment]
Out of curiosity, I just performed an experiment. I don't need a new key for a few days, but I bought a new one anyway. When the error happened a few minutes ago, I put in the new key. It had no effect. Not sure if this helps in any way, but I figured I'd share the results.

**Comment by lutra333** [comment]
Could confirm - i have 2 brand new keys, both of them having same ArgNull error.

**Comment by DarkLegend481** [comment]
For me, if sometimes I change my hash key, the game works again

**Comment by BadiBasso** [comment]
I believe that for some reason the saved accesstoken invalidates or gets corrupted between sessions.
Have tested by forcing app to request a new token at startup and for a couple days since I have not encountered the error.
Code I changed to test:
At GameClients.cs method CreateSession() changed:
AccessToken accessToken = GetAccessToken();
to
AccessToken accessToken = null;

**Comment by mtaheij** [comment]
Between sessions, the AccessToken can become outdated. The app should detect this and ask for another one.
I'm trying to include HockeyApp tracking for the next release, hoping to find answers to where things are going wrong.

**Comment by lutra333** [comment]
So, I downloaded the 1.3.3 entered brand new key an still have the same error. 

Deleted the app completely, restarted phone, installed 1.3.3 again - still the same error. 
![wp_ss_20170422_0001](https://cloud.githubusercontent.com/assets/27112334/25304885/00d7a882-2779-11e7-93d0-d9715ab358ea.png)

No new infromatiion on the bug unfortunately :((

**Comment by lutra333** [comment]
Everything started to work in a right wat in 1.3.4 after i use the button to 'forget' the access code. 
thanks!
